src/SphinxEngine.php

```diff
@@ -192,12 +192,18 @@ protected function performSearch(Builder $builder)
          */
         $model = $builder->model;
         $index = $model->searchableAs();
+        $ranker = config()->get('app.sphinx_ranker');
 
         $query = $this->sphinx
             ->select('*', SphinxQL::expr('WEIGHT() AS weight'))
             ->from($index)
-            ->match('*', SphinxQL::expr('"' . str_replace('"', '\"', $builder->query) . '"/1.0'))
-            ->limit($builder->limit??20);
+            ->match('*', SphinxQL::expr('"' . str_replace('"', '\"', $builder->query) . '"/1.0'));
+
+        if (!empty($ranker)) {
+            $query->option('ranker', $ranker);
+        }
```

Comment:
You can chain it like this as an alternative to if block: 
```
->when($ranker, function ($query, $ranker) {
        return $query->option('ranker', $ranker);
    })
 ```

Reply:
This package is dependent with Foolz\sphinxql-query-builder package and there is no `when` built in method inside Foolz\SphinxQL.php. Using when will result
<img width="1020" alt="Screen Shot 2023-07-26 at 12 20 39" src="https://github.com/rv-life/laravel-scout-sphinx/assets/111782856/e9b8e7ab-227b-4a07-b706-f25c73020524">